cylc/flow/tui/util.py

```diff
@@ -94,19 +107,17 @@ def compute_tree(flow):
     flow_node = add_node(
         'workflow', flow['id'], nodes, data=flow)
 
-    # create nodes
+    # populate cycle nodes
+    for cycle in flow['cyclePoints']:
+        cycle['id'] = idpop(cycle['id'])  # strip the family off of the id
+        cycle_node = add_node('cycle', cycle['id'], nodes, data=cycle)
+        flow_node['children'].append(cycle_node)
+
+    # populate family nodes
     for family in flow['familyProxies']:
```

Comment:
Is this loop here the same as the next loop further down? That iterates the same element, and calls `add_node` too? Could we now remove this one completely?

Reply:
Not quite, the first loop calls `add_node` with the `data` argument. Whereas the second loop calls it without.

This is because `add_node` is badly named, it's really `get_node`, the first loop creates the nodes, the second arranges them into the Cylc hierarchy.

This code is going to get heavily doctored when we move to deltas.